app/src/main/java/com/elmeradrianv/shesafe/fragments/MapFragment.java

```diff
@@ -0,0 +1,223 @@
+     * it inside the SupportMapFragment. This method will only be triggered once the user has
+     * installed Google Play services and returned to the app.
+     */
+    @Override
+    public void onMapReady(GoogleMap googleMap) {
+        map = googleMap;
+        map.setMapType(GoogleMap.MAP_TYPE_NORMAL);
+        getMyLocation();
+        queryReports();
+    }
+
+    @SuppressWarnings({"MissingPermission"})
+    @NeedsPermission({Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION})
+    void getMyLocation() {
+        map.setMyLocationEnabled(true);
+        map.getUiSettings().setMyLocationButtonEnabled(true);
+
+        FusedLocationProviderClient locationClient = getFusedLocationProviderClient(getContext());
+        locationClient.getLastLocation()
+                .addOnSuccessListener(location -> {
+                    if (location != null) {
+                        onLocationChanged(location);
+                    }
+                })
+                .addOnFailureListener(e -> {
+                    Log.d("MapDemoActivity", "Error trying to get last GPS location");
+                    e.printStackTrace();
+                });
+    }
+
+    public void onLocationChanged(Location location) {
+        // GPS may be turned off
+        if (location == null) {
+            return;
+        }
+        currentLocation = location;
+        displayLocation();
+    }
+
+    private void displayLocation() {
+        if (currentLocation != null) {
+            LatLng latLng = new LatLng(currentLocation.getLatitude(), currentLocation.getLongitude());
+            CameraUpdate cameraUpdate = CameraUpdateFactory.newLatLngZoom(latLng, 14);
+            map.animateCamera(cameraUpdate);
+        }
+    }
+
+
+    private List<Report> queryReports() {
+        List<Report> reports = new ArrayList<>();
+        ParseQuery<Report> query = ParseQuery.getQuery(Report.class);
+        query.include(Report.TYPE_OF_CRIME_KEY);
+        query.setLimit(50);
+        query.addDescendingOrder("createdAt");
+        query.findInBackground((reportList, e) -> {
+            if (e != null) {
+                Log.e(TAG, "Issue with getting posts", e);
+                return;
+            }
+            reports.addAll(reportList);
+            showReports(reports);
+        });
+        return reports;
+    }
+
+    private void showReports(List<Report> reports) {
+        for (Report report : reports) {
+            showMarker(report.getTypeOfCrime().getTag(),
+                    report.getLocation().getLatitude(),
+                    report.getLocation().getLongitude(),
+                    report.getTypeOfCrime().getLevelOfRisk());
+        }
+    }
+
+    private void showMarker(String title, double latitude, double longitude, int levelOfRisk) {
+        LatLng reportLatLng = new LatLng(latitude, longitude);
+        Marker marker;
```

Comment:
I recommend keeping initialization of variables as close as possible to their usage.

In this case Marker marker = map.addMarker(markerOptions); would be ideal.